### ISSUE TITLE
`final_exception': stack level too deep (SystemStackError)

### ISSUE THREAD

**Issue by lucianosousa** [rspec/rspec-core#2127]
This error is happening in all rspec-rails 3.x versions I tested here. since 3.0 until 3.4

`/Users/luciano/.rvm/gems/ruby-2.2.3/gems/rspec-core-3.4.1/lib/rspec/core/formatters/exception_presenter.rb:97:in `final_exception': stack level too deep (SystemStackError)`
https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/formatters/exception_presenter.rb#L97

note my ruby version. 2.2.3.
I teste in the ruby 2.1.3 and happen the same.

**Comment by lucianosousa** [comment]
I just removed the if/else test and works fine here:

before:
```
        def final_exception(exception)
          if exception.cause
            final_exception(exception.cause)
          else
            exception
          end
        end
```

after:

```
        def final_exception(exception)
          exception
        end
```

Any particular reason to this check?

**Comment by myronmarston** [comment]
The code is there to support the feature required in #2044 and implemented in #2074.  Getting infinite recursion is very odd here -- it suggests that you've got an exception for which `cause` returns itself.  I'm not sure how you'd get that.  Can you put together a repro so we can trouble shoot?  A simple fix could be written to only recurse if `exception.cause != exception`.
